### DISCUSSION TITLE
Handle SharePlugin share result, blocked by android native-bridge?

### DISCUSSION THREAD

**Discussion by kajk** [ionic-team/capacitor#5339] [Q&A] [answered]
I'm currently using the SharePlugin to use the native sharing feature, but when the user cancels the share an global error is thrown.

### Setup
  @capacitor/cli: 3.3.0
  @capacitor/core: 3.3.0
  @capacitor/android: 3.3.0
  @capacitor/ios: 3.3.1
  @capacitor/share:1.0.7
  @angular/core:12.2.1
Android Emulator (Android Version 11 and 12)

In an angular 12 App, calling the `SharePlugin.share({ title: ..., text: ... })` method, which results in the expected Sharing Popover to appear.
After clicking it away (not selecting a destination to share), my AngularErrorHandler is triggered with an error object looking as follows:
```
Error: Uncaught (in promise): Error: Share canceled
Error: Share canceled
    at Object.cap.fromNative ((index):413)
    at <anonymous>:1:18
    at resolvePromise (zone.js:1213)
    at zone.js:1120
    at zone.js:1136
    at ZoneDelegate.invoke (zone.js:372)
    at Object.onInvoke (core.js:28672)
    at ZoneDelegate.invoke (zone.js:371)
    at Zone.run (zone.js:134)
    at zone.js:1276
    at ZoneDelegate.invokeTask (zone.js:406)
    at Object.onInvokeTask (core.js:28659)
    at ZoneDelegate.invokeTask (zone.js:405)
    at Zone.runTask (zone.js:178)
    at drainMicroTaskQueue (zone.js:582)
```

This error seems to be triggered by the activityResult in the SharePlugin (See https://github.com/ionic-team/capacitor-plugins/blob/main/share/android/src/main/java/com/capacitorjs/plugins/share/SharePlugin.java#L44 ).
But my own calling component doesn't seem to be called with this error.

After further investigation the native-bridge ( https://github.com/ionic-team/capacitor/blob/1451ec850a9ef73267a032638e73f1fc440647b9/android/capacitor/src/main/assets/native-bridge.js#L413 ) seems to be handling this reject call, but I can't find out how that can be propagated further to my own code.

Has somebody else also once run into this issue? How did you resolve it?
Or am I just reading the fromNative method wrong, and there is a hook I've missed?
Any help / pointers would be greatly appreciated!

Cheers

**Comment by izeryab** [comment, marked as answer]
i was facing similar issue on Android 10 when i tried to share large text file using Share Plugin 
 
            capacitorFilesystem.Filesystem.getUri({
              directory: capacitorFilesystem.Directory.Data,
              path: "secret/large.text"
            })
          .then((uriResult) => {
           
            console.log( Capacitor.convertFileSrc(uriResult.uri));
             capacitorShare.Share.share({
              title: "Text File Share Test",
              text: "Large Text File",
              url: uriResult.uri,
            }).catch((e)=>{
              console.log("Error ",e)
            })
            
          }).catch((e)=>{
            console.log("Error ",e)
          })

**Reply by kajk** [reply]
Thanks a lot for your response! It helped plenty.

It seems as if the error really was in my code, and I simply missplaced the catch-clauses.
